compiler/src/dotty/tools/dotc/inlines/Inliner.scala

```diff
@@ -115,6 +116,18 @@ object Inliner:
       ConservativeTreeCopier()
     ):
 
+    override def transform(tree: Tree)(using Context): Tree =
+      tree match
+      case Typed(expr, tpt) =>
+        tpt.tpe match
+        case AnnotatedType(_, annot) if annot.hasSymbol(defn.NowarnAnnot) =>
```

Comment:
This is a small thing, but I believe we might encounter a typed tree with multiple annotations like `(ch: @nowarn @switch)` (I can't think of an example where this would be useful, but it is a possibility), so something like Typed(..., AnnotatedType(AnnotatedType(..., noWarnAnnot), )). In that case the nowarn might get ignored. it would be better to recurse on the type tree until we stop seeing `AnnotatedType`, like:

```suggestion
        def getNoWarnAnnot(tpe: Type): Option[Tree] = tpe match
          case AnnotatedType(_, annot) if annot.hasSymbol(defn.NowarnAnnot) => Some(annot)
          case AnnotatedType(underlying, _) => getNoWarnAnnot(underlying)
          case _ => None
        getNoWarnAnnot(tpt.tpe).foreach:
```

Reply:
thanks for the excellent spot!